packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -185,7 +185,29 @@ export interface TableProps {
   readonly enablePartitionFiltering?: boolean;
 
   /**
-   * The user-supplied properties for the description of the physical storage of this table.
+   * The user-supplied properties for the description of the physical storage of this table. These properties help describe the format of the data that is stored within the crawled data sources.
+   *
+   * There are reserved keys that are used by AWS Glue. They CAN be mutated, but they are best left alone.
+   *
+   * The key/value pairs that are allowed to be submitted are not limited, however their functionality is not guaranteed.
+   *
+   * @see https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_EXTERNAL_TABLE.html#r_CREATE_EXTERNAL_TABLE-parameters - under "TABLE PROPERTIES" contains a non-exhaustive list of the keys that have functionality.
+   *
+   * @example
+   *
+   *    declare const glueDatabase: glue.IDatabase;
+   *    const table = new glue.Table(this, 'Table', {
+   *      database: glueDatabase,
+   *      columns: [{
+   *          name: 'col1',
+   *          type: glue.Schema.STRING,
+   *      }],
+   *      dataFormat: glue.DataFormat.CSV,
+   *      storageParameters: {
+   *          foo: 'bar', // Will have no effect
```

Comment:
so this is a silent failure. If someone were to make a typo in their configuration, they'd have no way of knowing. I think we can improve this. We should define a `StorageParameters` enum like so:

```
export enum StorageParameters {
  SKIP_HEADER_LINE_COUNT = 'skip.header.line.count',
  DATA_CLEANSING_ENABLED = 'data_cleansing_enabled',
  // ...
}
```

We can define an enum for some of the values to as well:

```
export enum InvalidCharHandlingAction {
  DISABLED = 'DISABLED',
  FAIL = 'FAIL',
  // ...
  // This would hold all possible values, even though not every key supports every value. TS doesn't do enum inheritance and other languages don't do well with union types, so the best we can do here is runtime error checking. 
}
```

The compression type could use an enum as well:

```
export enum CompressionType {
  BZIP2 = 'bzip2',
  GZIP = 'gzip',
  // ...
}
```

Reply:
If someone wants to specify one we don't recognize, then they should be able to do so via:

```
StorageParameters.custom('unknown.property.thing');
```

Reply:
Regarding this, I've tried to implement the changes. You can see it within the code. 

- Firstly, does this look okay?
- Secondly, line 409 contains the annotation for the key/value pair. Assigning something like this
```ts
readonly storageParameters?: { [key in StorageParameters]?: string; };
```
Gives the following error
```
lib/table.ts:490:12 - error JSII1003: Only string-indexed map types are supported
```
Is there any way to annotate to use the mentioned `namespace`?

Reply:
I've now changed this to instead store `storageParameters` as an array, to alleviate this problem.